CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -696,4 +696,28 @@ public static function createDeferredTrxn($lineItems, $contributionDetails, $upd
     }
   }
 
+  /**
+   * Get Credit Card Details.
+   *
+   * @param int $contributionID
+   *   Contribution ID
+   *
+   * @return array
+   */
+  public static function getCreditCardDetails($contributionID, $viewOnly = TRUE) {
+    $sql = "SELECT card_type, pan_truncation
+      FROM civicrm_financial_trxn cft
+        INNER JOIN civicrm_entity_financial_trxn ceft ON ceft.financial_trxn_id = cft.id
+      WHERE ceft.entity_table = 'civicrm_contribution'
+        AND ceft.entity_id = {$contributionID}
```

Comment:
Just thinking rather than doing this could you do 
```
AND ceft.entity_id = %1
```

Then in L714 do 

```php
$dao = CRM_Core_DAO::executeQuery($sql, array(1 => array($contributionID, 'Positive')));
```